static/app/utils/makeSafeRefluxStore.ts

```diff
@@ -0,0 +1,50 @@
+export interface SafeRefluxStore extends Reflux.Store {
+  teardown(): void;
+  unsubscribeListeners: Reflux.Subscription[];
+}
+
+export function makeSafeRefluxStore<T extends Reflux.Store>(
```

Comment:
could we have a documentation comment above this to describe what it's doing?

This is interesting enough that we definitely should document what this is doing to make the reflux store "safe".